python/ray/tune/schedulers/pb2.py

```diff
@@ -0,0 +1,322 @@
+            specifies the distribution of a continuous parameter. You must
+            use tune.choice, tune.uniform, tune.loguniform, etc.. Arbitrary
+            tune.sample_from objects are not supported.
+            Tune will use the search space provided by
+            `hyperparam_mutations` for the initial samples if the
+            corresponding attributes are not present in `config`.
+        quantile_fraction (float): Parameters are transferred from the top
+            `quantile_fraction` fraction of trials to the bottom
+            `quantile_fraction` fraction. Needs to be between 0 and 0.5.
+            Setting it to 0 essentially implies doing no exploitation at all.
+        log_config (bool): Whether to log the ray config of each model to
+            local_dir at each exploit. Allows config schedule to be
+            reconstructed.
+        require_attrs (bool): Whether to require time_attr and metric to appear
+            in result for every iteration. If True, error will be raised
+            if these values are not present in trial result.
+        synch (bool): If False, will use asynchronous implementation of
+            PBT. Trial perturbations occur every perturbation_interval for each
+            trial independently. If True, will use synchronous implementation
+            of PBT. Perturbations will occur only after all trials are
+            synced at the same time_attr every perturbation_interval.
+            Defaults to False. See Appendix A.1 here
+            https://arxiv.org/pdf/1711.09846.pdf.
+
+    Example:
+        >>> pb2 = PB2(
+        >>>     time_attr="timesteps_total",
+        >>>     metric="episode_reward_mean",
+        >>>     mode="max",
+        >>>     perturbation_interval=10000,
+        >>>     hyperparam_mutations={
+        >>>         # These must be continuous, currently a limitation.
+        >>>         "factor_1": lambda: random.uniform(0.0, 20.0),
+        >>>     })
+        >>> tune.run({...}, num_samples=8, scheduler=pb2)
+    """
+
+    def __init__(self,
+                 time_attr: str = "time_total_s",
+                 reward_attr: Optional[str] = None,
+                 metric: Optional[str] = None,
+                 mode: Optional[str] = None,
+                 perturbation_interval: float = 60.0,
+                 hyperparam_mutations: Dict = None,
+                 quantile_fraction: float = 0.25,
+                 log_config: bool = True,
+                 require_attrs: bool = True,
+                 synch: bool = False):
+
+        resample_probability = 0
+        custom_explore_fn = None
+
+        super(PopulationBasedTraining, self).__init__(
+            self, time_attr, reward_attr, metric, mode, perturbation_interval,
+            hyperparam_mutations, quantile_fraction, resample_probability,
+            custom_explore_fn, log_config, require_attrs, synch)
+
+        self._name = "pb2"
```

Comment:
It doesn't seem like self._name is used anywhere. Is it safe to remove this?

Reply:
Yes, I added this with the intention of using it but then never did :)
Removed!